From 5baa7afdb688fcff8de12df0d67e084482aad86d Mon Sep 17 00:00:00 2001
From: Ma Wupeng <mawupeng1@huawei.com>
Date: Fri, 11 Nov 2022 09:32:24 +0800
Subject: [PATCH] mm: Refactor code in reliable_report_meminfo()
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 5baa7afdb688fcff8de12df0d67e084482aad86d
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4SK3S
CVE: NA

--------------------------------

Use show_val_kb() to format meminfo.

Signed-off-by: Ma Wupeng <mawupeng1@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/mem_reliable.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/mm/mem_reliable.c b/mm/mem_reliable.c
index fe6f47a0b4cd..d46fe86563bd 100644
--- a/mm/mem_reliable.c
+++ b/mm/mem_reliable.c
@@ -112,13 +112,19 @@ void shmem_reliable_init(void)
 		shmem_reliable = false;
 }
 
+static void show_val_kb(struct seq_file *m, const char *s, unsigned long num)
+{
+	seq_put_decimal_ull_width(m, s, num << (PAGE_SHIFT - 10), 8);
+	seq_write(m, " kB\n", 4);
+}
+
 void reliable_report_meminfo(struct seq_file *m)
 {
 	if (!mem_reliable_is_enabled())
 		return;
 
-	seq_printf(m, "ReliableTotal:    %8lu kB\n",
-		   total_reliable_mem_sz() >> 10);
-	seq_printf(m, "ReliableUsed:     %8lu kB\n",
-		   used_reliable_mem_sz() >> 10);
+	show_val_kb(m, "ReliableTotal:    ",
+			total_reliable_mem_sz() >> PAGE_SHIFT);
+	show_val_kb(m, "ReliableUsed:     ",
+			used_reliable_mem_sz() >> PAGE_SHIFT);
 }
-- 
2.33.0

